tests/Unit/Controller/ThreadControllerTest.php

```diff
@@ -0,0 +1,280 @@
+		$this->assertEquals(Http::STATUS_OK, $response->getStatus());
+	}
+
+	public function testDeleteTrash(): void {
+		$mailAccount = new MailAccount();
+		$mailAccount->setId(1);
+		$mailAccount->setTrashMailboxId(80);
+
+		$mailbox = new Mailbox();
+		$mailbox->setId(80);
+		$mailbox->setAccountId($mailAccount->getId());
+
+		$message = new Message();
+		$message->setId(300);
+		$message->setMailboxId($mailbox->getId());
+		$message->setThreadRootId('some-thread-root-id-1');
+
+		$this->mailManager
+			->method('getMessage')
+			->willReturn($message);
+
+		$this->mailManager
+			->method('getMailbox')
+			->willReturn($mailbox);
+
+		$this->accountService
+			->method('find')
+			->willReturn(new Account($mailAccount));
+
+		$this->threadMapper
+			->expects(self::once())
+			->method('findMessageUidsAndMailboxNamesByAccountAndThreadRoot')
+			->with($mailAccount, $message->getThreadRootId(), true)
+			->willReturn([
+				['messageUid' => 200, 'mailboxName' => 'Trash'],
+				['messageUid' => 300, 'mailboxName' => 'Trash'],
+			]);
+
+		$response = $this->controller->delete($message->getId());
+		$this->assertEquals(Http::STATUS_OK, $response->getStatus());
+	}
+
+	public function testMoveForbidden() {
+		$this->mailManager
+			->method('getMessage')
+			->willThrowException(new DoesNotExistException('for some reason there is no such record'));
+
+		$response = $this->controller->move(100, 20);
+		$this->assertEquals(Http::STATUS_FORBIDDEN, $response->getStatus());
+	}
+
+	public function testMoveInbox(): void {
+		$mailAccount = new MailAccount();
+		$mailAccount->setId(1);
+		$mailAccount->setTrashMailboxId(80);
+
+		$srcMailbox = new Mailbox();
+		$srcMailbox->setId(20);
+		$srcMailbox->setName('INBOX');
+		$srcMailbox->setAccountId($mailAccount->getId());
+
+		$dstMailbox = new Mailbox();
+		$dstMailbox->setId(40);
+		$dstMailbox->setName('Archive');
+		$dstMailbox->setAccountId($mailAccount->getId());
+
+		$message = new Message();
+		$message->setId(300);
+		$message->setMailboxId($srcMailbox->getId());
+		$message->setThreadRootId('some-thread-root-id-1');
+
+		$this->mailManager
+			->method('getMessage')
+			->willReturn($message);
+
+		$this->mailManager
+			->method('getMailbox')
+			->willReturnMap([
+				[$this->userId, $srcMailbox->getId(), $srcMailbox],
+				[$this->userId, $dstMailbox->getId(), $dstMailbox],
+			]);
+
+		$this->accountService
+			->method('find')
+			->willReturn(new Account($mailAccount));
+
+		$this->threadMapper
+			->expects(self::once())
+			->method('findMessageUidsAndMailboxNamesByAccountAndThreadRoot')
+			->with($mailAccount, $message->getThreadRootId(), false)
+			->willReturn([
+				['messageUid' => 200, 'mailboxName' => 'INBOX'],
+				['messageUid' => 300, 'mailboxName' => 'INBOX'],
+			]);
+
+		$response = $this->controller->move($message->getId(), $dstMailbox->getId());
+		$this->assertEquals(Http::STATUS_OK, $response->getStatus());
+	}
+
+	public function testMoveTrash(): void {
```

Comment:
tip for more readable test: structure them by the AAA principle, that is *A*rrange, *A*ct and *A*ssert. Keep those three as blocks without empty lines and then two lines between the three blocks. See https://github.com/nextcloud/mail/blob/c149af8e36d5491b2a9bb0133495f289e00d6945/tests/Unit/IMAP/Threading/ThreadBuilderTest.php#L73-L79 for a simple example.

Reply:
done ;)